libs/ui/checkbox/brain/src/lib/brn-checkbox.component.spec.ts

```diff
@@ -46,7 +46,7 @@ describe('BrnCheckboxComponent', () => {
 			`
      <label data-testid='label'>
      Checkbox Inside Label
-     <brn-checkbox disabled id='checkboxId' data-testid='checkbox' name='checkboxName'>
+     <brn-checkbox disabled='true' id='checkboxId' data-testid='checkbox' name='checkboxName'>
```

Comment:
Model by design don't have [transform](https://github.com/angular/angular/issues/55166#issuecomment-2032150999). I guess what we would need here is a writable input rather than two-way binding of the model input.

Best bet is to use a signal again for the state and change it from the input. What do you think?

```ts
private readonly _disabled = signal(false);
public readonly disabled = input(false, { transform: booleanAttribute });

constructor() {
       effect(
		() => {
				this._disabled.set(this.disabled());
		},
		{ allowSignalWrites: true },
	);
}
```

Reply:
If you change _disabled anywhere else, there might be differences in _disabled and disabled-values because its synchronized only in one direction. Previous non-signal inputs did not have this potential issue.

Reply:
We would use disabled only as input and reference _disabled everywhere else in the template and component.

Reply:
You can try this trick as well:
```ts
disabled = input(false, { transform: booleanAttribute });
disabledChange = outputFromObservable(toObservable(this.disabled));
```
It will work like a model with `transform` and you can't write to `disabledChange` directly (what @J216-H150 mentioned).

https://stackblitz.com/edit/stackblitz-starters-byydta?file=src%2Fmain.ts,src%2Fglobal_styles.css,node_modules%2F%40angular%2Fcore%2Frxjs-interop%2Findex.d.ts